pkg/security/probe/sysctl/snapshot.go

```diff
@@ -114,17 +114,21 @@ func (s *Snapshot) Snapshot(ignoredBaseNames []string, kernelCompilationFlags ma
 
 // snapshotProcSys recursively reads files in /proc/sys and builds a nested JSON structure.
 func (s *Snapshot) snapshotProcSys(ignoredBaseNames []string) error {
-	return filepath.Walk(kernel.HostProc("/sys"), func(file string, info fs.FileInfo, err error) error {
+	return filepath.WalkDir(kernel.HostProc("/sys"), func(file string, d fs.DirEntry, err error) error {
 		if err != nil {
 			return err
 		}
 
 		// Skip directories
-		if info.IsDir() {
+		if d.IsDir() {
 			return nil
 		}
 
 		// Skip if mode doesn't allow reading
+		info, err := d.Info()
+		if err != nil {
+			return nil // Skip files that can't be read
```

Comment:
Shouldn't you return `err` here to keep the same behavior as before ? (and similar in all 3 files owned by agent-security

Reply:
Good catch.
This should be fixed by [`e5cd39c` (#43982)](https://github.com/DataDog/datadog-agent/pull/43982/commits/e5cd39c489985ace4fea6856bcf26d8ddff6bdac).